datasets/flwr_datasets/resplitter/divide_resplitter_test.py

```diff
@@ -0,0 +1,228 @@
+# Copyright 2023 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""DivideResplitter tests."""
+
+import unittest
+from typing import Dict, Union
+
+from parameterized import parameterized_class
+
+from datasets import Dataset, DatasetDict
+from flwr_datasets.resplitter import DivideResplitter
+
+
+@parameterized_class(
+    ("divide_config", "divide_split", "drop_remaining_splits", "split_name_to_size"),
+    [
+        # Standard config that sums to one
+        (
+            {"train_1": 0.25, "train_2": 0.75},
+            "train",
+            False,
+            {"train_1": 10, "train_2": 30, "valid": 20, "test": 40},
+        ),
+        # As the first use case but drop the remaining splits
+        (
+            {"train_1": 0.25, "train_2": 0.75},
+            "train",
+            True,
+            {"train_1": 10, "train_2": 30},
+        ),
+        # Split does not sum to 1.0
+        (
+            {"a": 0.2, "b": 0.4},
+            "valid",
+            False,
+            {"a": 4, "b": 8, "train": 40, "test": 40},
+        ),
+        # Completely custom names
+        (
+            {"test_a": 0.2, "asdfasdfsa": 0.4},
+            "test",
+            False,
+            {"test_a": 8, "asdfasdfsa": 16, "valid": 20, "train": 40},
+        ),
+        # Mirror copies of the first example but using multiple split
+        (
+            {"train": {"train_1": 0.25, "train_2": 0.75}},
```

Comment:
would it be worth extending this to also resplit another split? (e.g. `valid`?)

Reply:
Ok, added now